utilities/rpp-unittests/HOST_NEW/BatchPD_host_pln3.cpp

```diff
@@ -3666,6 +3691,90 @@ int main(int argc, char **argv)
     {
         // test_case_name = "color_jitter";
         missingFuncFlag = 1;
+        break;
+    }
+    case 82:
+    {
+        test_case_name = "ricap";
+        unsigned int initial_permute_array[images];
+        Rpp32u permutedArray1[images] ;
+        Rpp32u permutedArray2[images] ;
+        Rpp32u permutedArray3[images] ;
+        Rpp32u permutedArray4[images] ;
+
+        Rpp32u cropRegion1[images] ;
+        Rpp32u cropRegion2[images] ;
+        Rpp32u cropRegion3[images] ;
+        Rpp32u cropRegion4[images] ;
+
+         
+        for (uint i = 0; i < images; i++ )
+        {
+            initial_permute_array[i] = i;
+        }
+        randomize(initial_permute_array, images);
+        for (uint i=0;i<images;i++)
+        {
+        permutedArray1[i] = (Rpp32u) initial_permute_array[i];
+        }
+        randomize(initial_permute_array, images);
+        for (uint i=0;i<images;i++)
+        {
+        permutedArray2[i] = (Rpp32u) initial_permute_array[i];
+        }
+        randomize(initial_permute_array, images);
+        for (uint i=0;i<images;i++)
+        {
+        permutedArray3[i] = (Rpp32u) initial_permute_array[i];
+        }
+        randomize(initial_permute_array, images);
+        for (uint i=0;i<images;i++)
+        {
+        permutedArray4[i] = (Rpp32u) initial_permute_array[i];
+        }
+
+ 
+        cropRegion1[0] = (Rpp32u) ((Rpp32f) 1) ; //x1
```

Comment:
Not sure why this double cast? Just a cropRegion1[0] = 1; should't do? Also, you've initialized cropRegion1[images] above (which could very well be batch size of 64/128), but need just [0-3]?